src/components/QuestionRow.tsx

```diff
@@ -0,0 +1,184 @@
+import { useEffect, useLayoutEffect, useRef, useState, type ClipboardEvent, type HTMLAttributes } from "react";
+import { z } from "zod";
+import { api } from "~/utils/api";
+import useDebounce from "~/utils/debounce";
+import { type Question } from "../types/global";
+import { parseMD, pasteText } from "../utils/utils";
+import { StyledCheckbox } from "./StyledCheckbox";
+import { StyledTextarea } from "./StyledInput";
+
+const MIN_TEXTAREA_HEIGHT_px = 41;
+const MAX_TEXTAREA_HEIGHT_px = MIN_TEXTAREA_HEIGHT_px * 2;
+const MAX_TEXTAREA_EDITING_HEIGHT_px = MIN_TEXTAREA_HEIGHT_px * 16;
+
+enum BODY_STATE {
+  COLLAPSED,
+  EXPANDED,
+  EDITING
+}
+const { COLLAPSED, EXPANDED, EDITING } = BODY_STATE;
+
+export const QuestionRow = ({
+  question, initialQuestion, onQuestionChange, onQuestionDelete, checked, indeterminate, ...others
+}: {
+  question: Question;
+  initialQuestion: Question;
+  onQuestionChange: (q: Question) => void;
+  onQuestionDelete?: () => void;
+  checked?: boolean;
+  indeterminate?: boolean;
+} & HTMLAttributes<HTMLDivElement>) => {
+
```

Comment:
```suggestion
small nit, might be a bit more readable
```suggestion
type QuestionRowProps = {
  question: Question;
  initialQuestion: Question;
  onQuestionChange: (q: Question) => void;
  onQuestionDelete?: () => void;
  checked?: boolean;
  indeterminate?: boolean;
} & HTMLAttributes<HTMLDivElement>;
export const QuestionRow = (props: QuestionRowProps) => {
  const { 
    question, initialQuestion, 
    onQuestionChange, onQuestionDelete, 
    checked, indeterminate, 
    ...others 
  } = props;
  const { title, body, difficulty, category } = question;
```
```